consent/consent-page/custom_otp_handler.go

```diff
@@ -0,0 +1,172 @@
+package main
+
+import (
+	"bytes"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"errors"
+	"io/ioutil"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+type CustomOTPHandler struct {
+	config OtpConfig
+	client *http.Client
+	Repo   *OTPRepo
+}
+
+func NewCustomOTPHandler(config Config, repo *OTPRepo) OTPHandler {
+	var (
+		pool = x509.NewCertPool()
+		bts  []byte
+		err  error
+	)
+
+	if bts, err = ioutil.ReadFile(config.RootCA); err != nil {
+		logrus.Errorf("failed to read root ca from %s", config.RootCA)
+		panic(err)
```

Comment:
I'd not use panic but return an error instead

Reply:
well, it is run on application init and we don't want to proceed if the root CA is missing, but maybe it will be cleaner to break execution in main file

Reply:
fixed